Userland/Libraries/LibWeb/WebIDL/AbstractOperations.h

```diff
@@ -63,4 +66,103 @@ JS::Completion construct(WebIDL::CallbackType& callback, Args&&... args)
+{
+    double upper_bound = 0;
+    double lower_bound = 0;
+
+    // 1. If bitLength is 64, then:
+    if constexpr (sizeof(T) == 4) {
+        // 1. Let upperBound be 2^(53) − 1
+        upper_bound = JS::MAX_ARRAY_LIKE_INDEX;
+
+        // 2. If signedness is "unsigned", then let lowerBound be 0.
+        if constexpr (IsUnsigned<T>) {
+            lower_bound = 0;
+        }
+        // 3. Otherwise let lowerBound be −2^(53) + 1.
+        else {
+            lower_bound = -JS::MAX_ARRAY_LIKE_INDEX;
+        }
+
+        // Note: this ensures long long types associated with [EnforceRange] or [Clamp] extended attributes are representable in ECMAScript’s Number type as unambiguous integers.
+    } else {
+        // 2. Otherwise, if signedness is "unsigned", then:
+        //     1. Let lowerBound be 0.
+        //     2. Let upperBound be 2^(bitLength) − 1.
+        // 3. Otherwise:
+        //     1. Let lowerBound be -2^(bitLength − 1).
+        //     2. Let upperBound be 2^(bitLength − 1) − 1.
+        lower_bound = NumericLimits<T>::min();
+        upper_bound = NumericLimits<T>::max();
+    }
+
+    // 4. Let x be ? ToNumber(V).
+    auto x = TRY(value.to_number(vm)).as_double();
+
+    // 5. If x is −0, then set x to +0.
+    if (x == -0.)
+        x = 0.;
+
+    // 6. If the conversion is to an IDL type associated with the [EnforceRange] extended attribute, then:
+    if (enforce_range == EnforceRange::Yes) {
+        // 1. If x is NaN, +∞, or −∞, then throw a TypeError.
+        if (isnan(x) || isinf(x))
+            return vm.throw_completion<JS::TypeError>(JS::ErrorType::NumberIsNaNOrInfinity);
+
+        // 2. Set x to IntegerPart(x).
+        x = integer_part(x);
+
+        // 3. If x < lowerBound or x > upperBound, then throw a TypeError.
+        if (x < lower_bound || x > upper_bound)
+            return vm.throw_completion<JS::RangeError>(MUST(String::formatted("Number '{}' is outside of allowed range of {} to {}", x, lower_bound, upper_bound)));
+
+        // 4. Return x.
+        return x;
+    }
+
+    // 7. If x is not NaN and the conversion is to an IDL type associated with the [Clamp] extended attribute, then:
+    if (clamp == Clamp::Yes && !isnan(x)) {
+        // 1. Set x to min(max(x, lowerBound), upperBound).
+        x = min(max(x, lower_bound), upper_bound);
+
+        // 2. Round x to the nearest integer, choosing the even integer if it lies halfway between two, and choosing +0 rather than −0.
+        // 3. Return x.
+        return round(x);
+    }
+
+    // 8. If x is NaN, +0, +∞, or −∞, then return +0.
+    if (isnan(x) || x == 0.0 || isinf(x))
+        return 0;
+
+    // 9. Set x to IntegerPart(x).
+    x = integer_part(x);
+
+    // 10. Set x to x modulo 2^bitLength.
+    auto const max_unsigned_value = AK::pow<double>(2, 8 * sizeof(T));
+    x = fmod(x, max_unsigned_value);
```

Comment:
I guess this should to be `JS::modulo`? It's actually giving me the incorrect answer (0) for something like:

`x = 1` and `2^bitLength` having bitlength of 64

Reply:
Fixed this issue :^) After the fix to `JS::modulo`, now able to use it without breaking the `Streams/ReadableByteStream-enqueue-respond.html` test